kernel/src/engine/ensure_data_types.rs

```diff
@@ -381,23 +386,26 @@ mod tests {
         )
         .is_ok());
 
-        assert!(ensure_data_types(
-            &DataType::Map(Box::new(MapType::new(
-                DataType::LONG,
-                DataType::STRING,
-                false
-            ))),
-            arrow_field.data_type(),
-            true
-        )
-        .is_err());
-
-        assert!(ensure_data_types(
-            &DataType::Map(Box::new(MapType::new(DataType::LONG, DataType::LONG, true))),
-            arrow_field.data_type(),
-            false
-        )
-        .is_err());
+        assert_result_error_with_message(
+            ensure_data_types(
+                &DataType::Map(Box::new(MapType::new(
+                    DataType::LONG,
+                    DataType::STRING,
+                    false,
+                ))),
+                arrow_field.data_type(),
+                true,
+            ),
+            "Generic delta kernel error: Map has nullablily false in kernel and true in arrow",
+        );
+        assert_result_error_with_message(
+            ensure_data_types(
+                &DataType::Map(Box::new(MapType::new(DataType::LONG, DataType::LONG, true))),
+                arrow_field.data_type(),
+                false,
+            ),
+            "Invalid argument error: Incorrect datatype. Expected long, got Utf8",
```

Comment:
in the future it would be nice to make this include context like it coming from a Map etc.? maybe make follow up issue?